src/shared/sam/activation.ts

```diff
@@ -305,54 +345,62 @@ async function promptInstallYamlPluginFromEditor(
+async function promptInstallYamlPluginFromFilename(fileName: string, disposables: vscode.Disposable[]): Promise<void> {
     if (fileName.endsWith('template.yaml') || fileName.endsWith('template.yml')) {
-        // immediately dispose other triggers so it doesn't flash again
-        for (const prompt of disposables) {
-            prompt.dispose()
-        }
-        const settings = PromptSettings.instance
+        promptInstallYamlPlugin(disposables)
+    }
+}
 
-        const installBtn = localize('AWS.missingExtension.install', 'Install...')
-        const permanentlySuppress = localize('AWS.message.info.yaml.suppressPrompt', "Don't show again")
+/**
+ * Show the install YAML extension prompt and dispose other listeners
+ * @param disposables
+ */
+async function promptInstallYamlPlugin(disposables: vscode.Disposable[]) {
+    // immediately dispose other triggers so it doesn't flash again
+    for (const prompt of disposables) {
+        prompt.dispose()
+    }
+    const settings = PromptSettings.instance
 
-        const response = await vscode.window.showInformationMessage(
-            localize(
-                'AWS.message.info.yaml.prompt',
-                'Install YAML extension for more {0} features in SAM template.yaml files.',
-                getIdeProperties().company
-            ),
-            installBtn,
-            permanentlySuppress
-        )
+    const installBtn = localize('AWS.missingExtension.install', 'Install...')
+    const permanentlySuppress = localize('AWS.message.info.yaml.suppressPrompt', "Don't show again")
 
-        switch (response) {
-            case installBtn:
-                // Available options are:
-                // extension.open: opens extension page in VS Code extension marketplace view
-                // workspace.extension.installPlugin: autoinstalls plugin with no additional feedback
-                // workspace.extension.search: preloads and executes a search in the extension sidebar with the given term
-
-                // not sure if these are 100% stable.
-                // Opting for `extension.open` as this gives the user a good path forward to install while not doing anything potentially unexpected.
-                try {
-                    await vscode.commands.executeCommand('extension.open', VSCODE_EXTENSION_ID.yaml)
-                } catch (e) {
-                    const err = e as Error
-                    getLogger().error(`Extension ${VSCODE_EXTENSION_ID.yaml} could not be opened: `, err.message)
-                }
-                break
-            case permanentlySuppress:
-                settings.disablePrompt('yamlExtPrompt')
-        }
+    const response = await vscode.window.showInformationMessage(
+        localize(
+            'AWS.message.info.yaml.prompt',
+            'Install YAML extension for more {0} features in SAM template.yaml files.',
```

Comment:
The logic implemented is not specific to SAM templates
```suggestion
            'Install YAML extension for more {0} features in CloudFormation templates',
```

Reply:
I think the original prompt message was just added as SAM because it's in the SAM folder. I'll change it